### DISCUSSION TITLE
Does subscribe("newBlockHeader",...) consistently provide the header of the most recent block?

### DISCUSSION THREAD

**Discussion by AlirezaEthDev** [ethereum/web3.js#6831] [Q&A]
Assuming we are trying the blow code that subscribes to incoming block headers and returns the header of the i-th block in the ledger, triggering the execution of the callback function in then(). If the callback function takes an extended period such that the (i+1)-th and (i+2)-th blocks are added to the ledger, will the subscriber receive the header of the (i+1)-th or (i+2)-th block for subscription? Does it consistently provide the header of the most recent block?

```
web3.eth.subscribe("newBlockHeaders",function(error,result){
    if(error){
        throw error;
    }else{
        console.info("Result:\n"+result);
    }
}).then(function(subscription){
    subscription.on("data",function(blockHeader){
...
       })
})
```

**Comment by TatyOko28** [comment]
No, subscribe("newBlockHeaders") does not guarantee delivery of every block header if the callback execution time is too long.

 The subscriber will always receive the latest block header, but it might skip intermediate blocks if the callback execution takes too long.
 If you need every block header, consider using a queue or a dedicated event listener.
